### PR TITLE
PWGHF: add muon analysis task

### PR DESCRIPTION
**Pull request by MaolinZH**

### CODE REVIEW

**Comment by MaolinZH** [comment]
> 

This task is derived from the DQ framework and it is used to extract the observables on single muons (pT, eta, sign of the muon, DCA,  etc) needed for the HF-muon analysis. It is indeed linked to HF since the final goal is to measure muons from charm decays and muons from beauty decays, separately from the information estracted with this task

**Comment by vkucera** [comment]
Hi @MaolinZH , thanks for the clarification. I suppose your code should be moved into the HFL directory then.
Your code is not passing the tests. Fix the errors first.

File: PWGHF/Tasks/HFmAnalysis.cxx [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+#include "Common/DataModel/EventSelection.h"
+#include <TH1F.h>
+#include <TH2I.h>
+#include <THashList.h>
+#include <TString.h>
+#include <vector>
+#include <memory>
+#include <cstring>
+#include <THnSparse.h>
+#include "TDatabasePDG.h"
```
Comment (vkucera) [inline PWGHF/Tasks/HFmAnalysis.cxx:24]: I think you have way more includes than needed.

**Review by vkucera — CHANGES_REQUESTED** [review]
Check the naming conventions.
Add proper documentation.

**Review by vkucera — CHANGES_REQUESTED** [review]

File: PWGHF/Tasks/HFmAnalysis.cxx [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+  }
+//=============================================================================
+  template <uint32_t TEventFillMap, uint32_t TMuonFillMap, typename TEvent, typename TMuons>
+  void runDataMuonSel(TEvent const &event, aod::BCs const &bcs, TMuons const &tracks) {
+    if(event.isEventSelected()==0) return;
+
+    VarManager::ResetValues(0,VarManager::kNMuonTrackVariables,fValues);
+    VarManager::FillEvent<gkEventFillMap>(event,fValues);
+
+    for(auto &track: tracks) {
```
Comment (vkucera) [inline PWGHF/Tasks/HFmAnalysis.cxx:184]:
```suggestion
    for(auto const& track: tracks) {
```
I suppose you don't want to change the tracks.

File: PWGHF/Tasks/HFmAnalysis.cxx [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+
+      if (fTrackCut->IsSelected(fValues)) registry.fill(HIST("muAcuts"),
+        fValues[VarManager::kPt],
+        fValues[VarManager::kEta], dcaXY,
+        fValues[VarManager::kCharge], track.p(),
+	fValues[VarManager::kVtxZ],
+        fValues[VarManager::kMuonTrackType],0);	
+	}
+  }
+//=============================================================================
```
Comment (vkucera) [inline PWGHF/Tasks/HFmAnalysis.cxx:219]: Remove these lines everywhere.

File: PWGHF/Tasks/HFmAnalysis.cxx [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+
+
+  template <uint32_t TEventFillMap, uint32_t TMuonFillMap, uint32_t TTrackMCFillMap, typename TEvent, typename TMuons, typename TMC>
+  void runMCMuonSel(TEvent const &event, aod::BCs const &bcs, TMuons const &tracks, TMC const &mc) {
+    if(event.isEventSelected()==0) return;
+
+    VarManager::ResetValues(0,VarManager::kNMuonTrackVariables,fValues);
+    VarManager::FillEvent<gkEventFillMap>(event,fValues);
+
+    for(auto &track: tracks) {
```
Comment (vkucera) [inline PWGHF/Tasks/HFmAnalysis.cxx:229]:
```suggestion
    for(auto const& track: tracks) {
```

**Review by vkucera — CHANGES_REQUESTED** [review]

**Comment by MaolinZH** [comment]
> Hi @MaolinZH , thanks for the clarification. I suppose your code should be moved into the HFL directory then. Your code is not passing the tests. Fix the errors first.

Thank you for the comments, I will fix the errors and format, and commit a new version of the code.

**Comment by vkucera** [comment]
Hi @MaolinZH , can you please try to update this PR with the latest version of your branch to refresh the tests?

**Comment by vkucera** [comment]
Replaced with https://github.com/AliceO2Group/O2Physics/pull/1281

**Pull request closed without merging**